tests/test_unit_oob.cpp

```diff
@@ -166,9 +175,89 @@ void test_oob(void **) {
     }
 }
 
+// Tests DSN logging in OOB
+void test_dsn(void **) {
+    int evnt = 0;
+
+    std::string CABundlePath = getCABundleFile();
+
+    const std::vector<std::string> SF_SENSITIVE_KEYS = std::vector<std::string>{"UID", "PWD", "TOKEN", "PASSCODE", "PRIV_KEY_FILE_PWD"};
+
+    SF_DSN dsnParameters[] = {
+            {"SERVER", "snowflake.local.snowflakecomputing.com"},
+            {"PORT", "443"},
+            {"ACCOUNT", "testaccount"},
+            {"Uid", "snowman"},
+            {"PWD", "dummy"},
+            {"Token", "TOKEN"},
+            {"Authenticator", "snowflake"},
+            {"DATABASE", "TestDB"},
+            {"SCHEMA", "TestSchema"},
+            {"WAREHOUSE", "Regress"},
+            {"ROLE", "AccountAdmin"},
+            {"SSL", "1"},
+    };
+
+    int count = sizeof(dsnParameters)/sizeof(SF_DSN);
+    struct dsnKeyValue* kvPairs = (struct dsnKeyValue*) malloc(sizeof(struct dsnKeyValue)*count);
+
+    for (int i = 0; i < count; ++i) {
+        kvPairs[i] = dsnKeyValue{dsnParameters[i].key, dsnParameters[i].val};
+        for (int j = 0; j < SF_SENSITIVE_KEYS.size(); ++j) {
+            if (!strcasecmp(dsnParameters[i].key, SF_SENSITIVE_KEYS[j].c_str())) {
+                kvPairs[i] = dsnKeyValue{dsnParameters[i].key, "***"};
+                break;
+            }
+        }
+    }
+    setOOBDsninfo(kvPairs, count);
+
+    char connStr[] = "/session/v1/login-request?requestId=c4d53986-ee7a-4f01-9fac-2604653e9c41&request_guid=abbab0e5-5c77-4102-9d29-d44efde6a050&databaseName=testdb&schemaName=testschema&warehouse=regress";
+    char url[1024] = {0};
+    sprintf(url, "%s:%s%s", dsnParameters[0].val, dsnParameters[1].val, connStr);
+    setConnectionString((char const *) url);
+
+    setOOBeventdata(OOBEVENTNAME, (const char*)"ConnectionDSNConfig", 0);
+    setOOBeventdata(ERRORCODE, NULL, 0);
+    setOOBeventdata(EXCPMSG, "", 0);
+    setOOBeventdata(REQUESTURL, url, 0);
+    setOOBeventdata(OOBSQLSTATE, (const char*)"", 0);
+    setOOBeventdata(OOBCABUNDLE, CABundlePath.c_str(), 0);
+    char *oobevent = prepareOOBevent(nullptr);
+
```

Comment:
add an assertion that once the event has been prepared, the dsn object should be freed.